scipy/stats/mstats_basic.py

```diff
@@ -2598,8 +2598,10 @@ def friedmanchisquare(*args):
         ranked = ranked._data
     (k,n) = ranked.shape
     # Ties correction
-    repeats = np.array([find_repeats(_) for _ in ranked.T], dtype=object)
-    ties = repeats[repeats.nonzero()].reshape(-1,2)[:,-1].astype(int)
+    repeats = [find_repeats(_) for _ in ranked.T]
+    ties = [(x, y) for x, y in repeats if x.size > 0]
+    ties = np.array(ties, dtype=object).reshape(-1, 2)[:,-1].astype(int)
```

Comment:
Why is the object array here in the first place? Can't this become `ties = np.array([y for x, y in repeats if x.size > 0])`?

Reply:
Yeah, and that reshape isn't necessary any more either, and nor is the `astype(int)` if you just do

`ties = np.array([y for x, y in repeats if x.size > 0])`

Got a little too carried away with making only minimal changes I guess.